### DISCUSSION TITLE
will ts-node resolve export patterns with multiple specifiers?

### DISCUSSION THREAD

**Discussion by iambumblehead** [TypeStrong/ts-node#1877] [Q&A]
hello, I'm updating a package that does currently work with ts-node using this simple export pattern BUT...
```json
{
  "exports": {
    "types": "./src/esmock.d.ts",
    "import": "./src/esmockLoader.js"
  }
}
```

When updating the export pattern to define multiple specifiers, ts-node does not resolve these and, instead, this error appears "error TS2307: Cannot find module 'esmock/strict' or its corresponding type declarations."
```json
{
  "exports": {
    ".": {
      "import": {
        "types": "./src/esmock.d.ts",
        "default": "./src/esmockLoader.js"
      }
    },
    "./strict": {
      "import": {
        "types": "./src/esmock.d.ts",
        "default": "./src/esmockStrict.js"
      }
    },
    "./partial": {
      "import": {
        "types": "./src/esmock.d.ts",
        "default": "./src/esmockPartial.js"
      }
    }
  }
}
```

```json
{
  "exports": {
    ".": {
      "types": "./src/esmock.d.ts",
      "import": "./src/esmockLoader.js"
    },
    "./strict": {
      "types": "./src/esmock.d.ts",
      "import": "./src/esmockStrict.js"
    },
    "./partial": {
      "types": "./src/esmock.d.ts",
      "import": "./src/esmockPartial.js"
    }
  }
}
```

Would someone tell me what I'm doing wrong and suggest ways to solve this problem? A PR with the failing ts-node tests is here https://github.com/iambumblehead/esmock/pull/139 a related "announcement" about typescript and esm export patterns is here https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing

I am a novice typescript user --thank you in advance for any help or advice you might give

**Comment by koshic** [comment]
It's a typical ts-node problem - it provides only pain without complicated configuration required by typescript. TS work fine with package exports with conditions, nested conditions and even with package imports:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/20106607/196404503-7e73285a-e236-4769-ad4a-7f923f7870dd.png">

Could you please provide a bit more detail about TS setup in your tests? 

Example of my strictness tsconfig for node 18 + ESM:
```
{
  "compilerOptions": {
    "allowUnusedLabels": false,
    "allowUnreachableCode": false,
    "esModuleInterop": true,
    "exactOptionalPropertyTypes": true,
    "forceConsistentCasingInFileNames": true,
    "importsNotUsedAsValues": "error",
    "isolatedModules": true,
    "lib": ["ESNext"],
    "module": "NodeNext",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noPropertyAccessFromIndexSignature": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "strict": true,
    "skipLibCheck": true,
    "target": "ESNext",
    "types": []
  }
}
```
